wasm/main.py

```diff
@@ -196,27 +201,17 @@ def spec_validate_functype(ft):
 
 
 def spec_validate_tabletype(table_type: TableType) -> TableType:
-    # TODO: use of UINT32_CEIL may be incorrect here as `validate_limit` checks
-    # against the value using `>`.  Check looks like it **should** be using
-    # `>=` to ensure that the limit bounds fit withint a UINT32 but updating
-    # this to use `UINT32_MAX` is currently causing spec test failures.
-    # TODO: The type casting to `UInt32` is technically incorrect here since
-    # `UINT32_CEIL` is outside the valid range for u32, but we first need to
-    # resolve the above TODO.
```

Comment:
cc @carver I resolved this.

The `k` variable, now referred to as `upper_bound` is intended to be an inclusive upper bound as per the spec, but **also** there was some potentially missing validation on the general constraints of the `limits.max` and `limits.min` that I added to `spec_validate_limits` but it's also worth noting that this validation is duplicated in [`spec_binary_uN`](https://github.com/ethereum/py-wasm/blob/2fa5a06d59d456053002cfc79bf7bb71f93c8245/wasm/main.py#L3318-L3325) which is used to extract the values that become `limits.min` and `limits.max` so at some point this should be consolidated.